### ISSUE TITLE
PHP Warning:  Cannot use a scalar value as an array in DataObject.inc.php on line 133

### ISSUE THREAD

**Issue by hekutoru2kx** [pkp/pkp-lib#2772]
Greetings,

We recently updated from 2.4.8.2 to 3.0.2. We are a multi journal site (with some multilingual journals) and we are experiencing some functionality issues due to a warning php error. Our database is a postgres database.

There a two actions that can't be make specifically:
1. Consulting the submissions (there is no submission options)
2. Consulting the details of back issues (it list the issues but I can't access them)

Both of this actions are showing the error (This error also appears in other parts of the site but so far doesn't affected the way it works)
`PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/xxxxxx`

This is the log error (without debugging option active):
`[Thu Sep 07 01:28:26.458290 2017] [:error] [pid 22092] [client 181.33.70.56:12869] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222, referer: http://revistas.uis.edu.co/index.php/revistaprueba/submissions
[Thu Sep 07 01:28:27.161496 2017] [:error] [pid 22092] [client 181.33.70.56:12869] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/submissions
[Thu Sep 07 01:37:46.290907 2017] [:error] [pid 22511] [client 181.33.70.56:13040] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues
[Thu Sep 07 01:37:46.290999 2017] [:error] [pid 22511] [client 181.33.70.56:13040] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues`

This is the log error (with debugging):
`[Thu Sep 07 01:28:27.161496 2017] [:error] [pid 22092] [client 181.33.70.56:12869] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/submissions
[Thu Sep 07 01:28:27.161632 2017] [:error] [pid 22092] [client 181.33.70.56:12869] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/submissions
[Thu Sep 07 01:28:29.690515 2017] [:error] [pid 22088] [client 66.249.64.218:57859] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:28:29.909955 2017] [:error] [pid 22088] [client 66.249.64.218:57859] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 1825). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:28:29.937667 2017] [:error] [pid 22088] [client 66.249.64.218:57859] PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/revista/lib/pkp/lib/adodb/adodb.inc.php:519) in /var/www/html/revista/lib/pkp/classes/template/PKPTemplateManager.inc.php on line 806
[Thu Sep 07 01:28:29.937704 2017] [:error] [pid 22088] [client 66.249.64.218:57859] PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/revista/lib/pkp/lib/adodb/adodb.inc.php:519) in /var/www/html/revista/lib/pkp/classes/template/PKPTemplateManager.inc.php on line 807
[Thu Sep 07 01:28:30.353251 2017] [:error] [pid 22155] [client 40.77.167.58:8578] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222`

`[Thu Sep 07 01:46:41.752294 2017] [:error] [pid 22534] [client 181.33.70.56:9135] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues
[Thu Sep 07 01:46:41.754150 2017] [:error] [pid 22513] [client 181.33.70.56:9136] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues
[Thu Sep 07 01:46:42.311245 2017] [:error] [pid 22512] [client 181.33.70.56:9134] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues
[Thu Sep 07 01:46:42.311379 2017] [:error] [pid 22512] [client 181.33.70.56:9134] PHP Warning:  Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133, referer: http://revistas.uis.edu.co/index.php/revistaprueba/manageIssues
[Thu Sep 07 01:46:48.495132 2017] [:error] [pid 22667] [client 66.249.64.211:44416] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:47:01.414253 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:47:01.614043 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 1578). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:01.678685 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 1578). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:01.686501 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 1578). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:01.690984 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:47:01.709041 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/revista/lib/pkp/lib/adodb/adodb.inc.php:519) in /var/www/html/revista/lib/pkp/classes/template/PKPTemplateManager.inc.php on line 806
[Thu Sep 07 01:47:01.709074 2017] [:error] [pid 22520] [client 157.55.39.35:7880] PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/revista/lib/pkp/lib/adodb/adodb.inc.php:519) in /var/www/html/revista/lib/pkp/classes/template/PKPTemplateManager.inc.php on line 807
[Thu Sep 07 01:47:03.546423 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:47:03.751246 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 2708). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:03.816979 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 2708). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:03.823932 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Warning:  pg_query_params(): Query failed: ERROR:  llave duplicada viola restricci\xc3\xb3n de unicidad \xc2\xabcontrolled_vocab_symbolic\xc2\xbb\nDETALLE:  Ya existe la llave (symbolic, assoc_type, assoc_id)=(submissionKeyword, 1048585, 2708). in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 228
[Thu Sep 07 01:47:03.828682 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Notice:  Undefined offset: 0 in /var/www/html/revista/lib/pkp/lib/adodb/drivers/adodb-postgres7.inc.php on line 222
[Thu Sep 07 01:47:03.847056 2017] [:error] [pid 22511] [client 66.249.64.211:41690] PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/revista/lib/pkp/lib/adodb/adodb.inc.php:519) in /var/www/html/revista/lib/pkp/classes/template/PKPTemplateManager.inc.php on line 806`

The closest related issue that I could find in the forum is this https://github.com/pkp/pkp-lib/issues/1336 but I couldn't understand the fix (delete the offending no-locale settings from issue_settings).

Thanks for the good work

**Comment by asmecher** [comment]
@hekutoru2kx, have you worked through the discussions of this issue in the forum? See https://forum.pkp.sfu.ca/search?q=Cannot%20use%20a%20scalar%20value%20as%20an%20array -- I think there are a few checks/solutions there.

**Comment by hekutoru2kx** [comment]
Hi Alec,

Yes, I found no solution there.

I checked the issues and issue_settings and there is so far no no-localized settings.
`ojs=> select * from issue_settings where locale is null;
 issue_id | locale | setting_name | setting_value | setting_type 
----------+--------+--------------+---------------+--------------
(0 filas)

ojs=> select * from issue_settings where trim(locale) = '';
 issue_id | locale |     setting_name     | setting_value | setting_type 
----------+--------+----------------------+---------------+--------------
      467 |        | pub-id::publisher-id |               | string
      496 |        | pub-id::publisher-id |               | string
      553 |        | pub-id::publisher-id |               | string
      571 |        | pub-id::publisher-id |               | string
      556 |        | pub-id::publisher-id |               | string
      576 |        | pub-id::publisher-id |               | string
(6 filas)
`


I edited the line 133 to show the status of the variables:
`if (!is_array($this->_data[$key])) {
		error_log(print_r($key,true));
		error_log(print_r($this->_data[$key],true));
	}
	$this->_data[$key][$locale] = $value;`

Result is attached
[ErrorLog.txt](https://github.com/pkp/pkp-lib/files/1285608/ErrorLog.txt)

**Comment by hekutoru2kx** [comment]
Greetings,

I rolled back the database changes to my 2.3.8 and did the update with the official release with several patches (not the github as before) and the errors of tsubmission and back issues dissapeared. Still, the warning remains but is not related.

PHP Warning: Cannot use a scalar value as an array in /var/www/html/revista/lib/pkp/classes/core/DataObject.inc.php on line 133

I'm currently solving some issues with the forum recommendations.

According to all the post read and some test that I did updating from git in different dates, it seems like the github master suffered some kind of rollbacks that mismatched the database settings. Anyway, this is all part of the evolution.

Thank you
